src/test/DictionaryManager.java

```diff
@@ -21,11 +21,36 @@ public static DictionaryManager get() {
     }
 
     public boolean query(String... args) {
-        return false;
+        Boolean ret = false;               
+        String word = args[args.length - 1]; // Save the last arg in a variable
+        for (int i = 0; i < args.length - 1; i++) { // Go over all the args except the last one
+            String fn = args[i];
+            Dictionary dic = new Dictionary(fn);
+            this.dictionaryMap.put(fn, dic);
+        }
+        for (Dictionary value : dictionaryMap.values()) {
+            if (value.query(word)){

```

Comment:
`dictionary.close()` after the search ends (otherwise the searcher threads keep running)